builder/support.go

```diff
@@ -1,9 +1,18 @@
 package builder
 
 import (
+	"regexp"
 	"strings"
 )
 
+const acceptableRemoteMIME = `(?:application/(?:(?:x\-)?tar|octet\-stream|((?:x\-)?(?:gzip|bzip2?|xz)))|(?:text/plain))`
+
+var mimeRe = regexp.MustCompile(acceptableRemoteMIME)
+
+func selectAcceptableMIME(ct string) string {
+	return mimeRe.FindString(ct)
```

Comment:
@tiborvass  `Content-Type` values for a given encoding format can show up in one of a few variants. For example, bzip encoding can be signaled with either `application/bzip`, `application/x-bzip` or `application/bzip2`; similarly for gzip, xz, etc. I think a regex is the least verbose way of handling that kind of variation.